services/datamanager/builtin/builtin.go

```diff
@@ -610,28 +614,82 @@ func (svc *builtIn) Reconfigure(
 		!reflect.DeepEqual(svc.tags, svcConfig.Tags) || svc.fileLastModifiedMillis != fileLastModifiedMillis ||
 		svc.maxSyncThreads != newMaxSyncThreadValue
 
 	if syncConfigUpdated {
+		svc.syncConfigUpdated = syncConfigUpdated
 		svc.syncDisabled = svcConfig.ScheduledSyncDisabled
 		svc.syncIntervalMins = svcConfig.SyncIntervalMins
 		svc.tags = svcConfig.Tags
 		svc.fileLastModifiedMillis = fileLastModifiedMillis
 		svc.maxSyncThreads = newMaxSyncThreadValue
+	}
+
+	// if datacapture is enabled, kick off a go routine to check if disk space is filling due to
+	// cached datacapture files
+	if !svc.captureDisabled {
+		fileDeletionCtx, cancelFunc := context.WithCancel(context.Background())
+		svc.fileDeletionRoutineCancelFn = cancelFunc
+		svc.fileDeletionBackgroundWorkers = &sync.WaitGroup{}
+		svc.fileDeletionBackgroundWorkers.Add(1)
+		go pollFilesystem(fileDeletionCtx, svc.fileDeletionBackgroundWorkers,
+			svc.captureDir, deleteEveryNthValue, svc.syncer, svc.logger)
+	}
+
+	g.Success()
+	return nil
+}
+
+func (svc *builtIn) startPropagateDataSyncConfig() {
+	svc.dataCaptureWorkers.Add(1)
+	goutils.ManagedGo(svc.propagateDataSyncConfigLoop, svc.dataCaptureWorkers.Done)
+}
 
+// propagateDataSyncConfigLoop runs until Close() is called on *builtIn
+// Immediately on first execution and every second afterwards it
+// checks if the datasync configuration has changes which
+// have not propagated to datasync.
+// If so it propagates the changes and marks the datasync configuration as propagated.
+// Otherwise it sleeps for another second.
+// Takes the builtIn lock every iteration.
+func (svc *builtIn) propagateDataSyncConfigLoop() {
+	if err := svc.propagateDataSyncConfig(); err != nil {
+		return
+	}
+	for goutils.SelectContextOrWait(svc.closedCtx, time.Second) {
+		if err := svc.propagateDataSyncConfig(); err != nil {
+			return
+		}
+	}
+}
+
+func (svc *builtIn) propagateDataSyncConfig() error {
+	svc.lock.Lock()
+	if svc.syncConfigUpdated {
 		svc.cancelSyncScheduler()
 		if !svc.syncDisabled && svc.syncIntervalMins != 0.0 {
 			if svc.syncer == nil {
-				if err := svc.initSyncer(ctx); err != nil {
-					return err
+				if err := svc.initSyncer(svc.closedCtx); err != nil {
+					svc.lock.Unlock()
+					if errors.Is(err, cloud.ErrNotCloudManaged) {
+						svc.logger.Debug("Using no-op sync manager when not cloud managed")
+						return err
+					}
+					svc.logger.Infof("initSyncer err: %s", err.Error())
+					return nil
 				}
-			} else if reinitSyncer {
+			} else if svc.reinitSyncer {
 				svc.closeSyncer()
-				if err := svc.initSyncer(ctx); err != nil {
-					return err
+				if err := svc.initSyncer(svc.closedCtx); err != nil {
+					svc.lock.Unlock()
+					if errors.Is(err, cloud.ErrNotCloudManaged) {
+						svc.logger.Debug("Using no-op sync manager when not cloud managed")
+						return err
+					}
+					svc.logger.Infof("initSyncer err: %s", err.Error())
+					return nil
```

Comment:
is this the same as L674-681 with the addition of `svc.CloseSyncer`? could we just DRY and have that close in a small `if` condition?

Reply:
Yes it is the same however I don't see how I can dry this up without changing the behavior giv
en we need to propagate the initialization error in one case to terminate the loop, return no error in another case to continue the loop and continue the function in another case.

I'd also like to leave drying up to later PRs which are focused on reducing code complexity. I'm trying to keep this PR to the minimal change required to fix the current issues.